File: haohaninfo/即時範例/function.py
# -*- coding: UTF-8 -*-
#載入相關套件
import time
import datetime
import tailer

#取得當天日期
Date=time.strftime("%Y%m%d")
#設定檔案位置
DataPath="D:/data/"

#開啟這三個檔案
MatchFile=open(DataPath+Date+'_Match.txt')
OrderFile=open(DataPath+Date+'_Commission.txt')
UpDn5File=open(DataPath+Date+'_UpDn5.txt')

#持續取得成交資訊
def getMatch():
 return tailer.follow(MatchFile,0)

#持續取得委託資訊
def getOrder():
 return tailer.follow(OrderFile,0)

#持續取得上下五檔價資訊
def getUpDn5():
 return tailer.follow(UpDn5File,0)

#取得最新一筆成交資訊
def getLastMatch():
 return tailer.tail(MatchFile,3)[-2].split(",")

#取得最新一筆委託資訊
def getLastOrder():
 return tailer.tail(OrderFile,3)[-2].split(",")

#取得最新一筆上下五檔價資訊
def getLastUpDn5():
 return tailer.tail(UpDn5File,3)[-2].split(",")

File: haohaninfo/交易實務案例/function.R
#設置當天日期，取用當天的檔案名稱
Date <- gsub("-","",Sys.Date())
#檔案位置
DataPath <- "D:/data/"
#tail執行檔位置
tailPath <- "./bin/"

#取得成交資訊
GetMatchData <- function(DataPath,Date)
{

 data <- system(paste0(tailPath,'tail.exe -n1 ',DataPath,Date,"_Match.txt"),intern=TRUE)
 mdata <- strsplit(data,",")
 return(mdata)

}

#取得委託資訊
GetOrderData <- function(DataPath,Date)
{

 data <- system(paste0(tailPath,'tail.exe -n1 ',DataPath,Date,"_Commission.txt"),intern=TRUE)
 mdata <- strsplit(data,",")
 return(mdata)

}

#取得上下五檔價資訊
GetUpDn5Data <- function(DataPath,Date)
{

 data <- system(paste0(tailPath,'tail.exe -n1 ',DataPath,Date,"_UpDn5.txt"),intern=TRUE)
 mdata <- strsplit(data,",")
 return(mdata)

}
